### PR TITLE
FOGL-3480: Fixed subscription memory leak issue

### PR DESCRIPTION
**Pull request by shariq1501**

### CODE REVIEW

File: src/app/components/layout/navbar/navbar.component.ts [hunk 1]
```diff
@@ -293,10 +294,6 @@ export class NavbarComponent implements OnInit, AfterViewInit, OnDestroy {
     clearInterval(this.timer);
   }
 
-  ngOnDestroy() {
-    clearInterval(this.timer);
```
Comment (praveen-garg) [inline src/app/components/layout/navbar/navbar.component.ts:297]: is this happening via super?

File: src/app/components/core/dashboard/dashboard.component.ts [hunk 2]
```diff
@@ -255,8 +257,4 @@ export class DashboardComponent implements OnInit, OnDestroy {
     this.showGraph(data);
     this.toggleDropDown(id);
   }
-
-  public ngOnDestroy(): void {
-    this.isAlive = false;
```
Comment (praveen-garg) [inline src/app/components/core/dashboard/dashboard.component.ts:260]: is this also being taken care of by super?

File: src/app/components/core/logs/list-tasks/list-tasks.component.ts [hunk 3]
```diff
@@ -99,8 +101,4 @@ export class ListTasksComponent implements OnInit, OnDestroy {
         }
       });
   }
-
-  public ngOnDestroy(): void {
-    this.isAlive = false;
```
Comment (praveen-garg) [inline src/app/components/core/logs/list-tasks/list-tasks.component.ts:104]: is this also being taken care of by super?

File: src/app/components/core/north/north.component.ts [hunk 4]
```diff
@@ -107,11 +107,5 @@ export class NorthComponent implements OnInit, OnDestroy {
   public hideLoadingSpinner() {
     this.showSpinner = false;
   }
-
-  public ngOnDestroy(): void {
-    this.isAlive = false;
```
Comment (praveen-garg) [inline src/app/components/core/north/north.component.ts:112]: is this also being taken care of by super?

File: src/app/components/core/asset-readings/assets/assets.component.ts [hunk 5]
```diff
@@ -138,14 +140,10 @@ export class AssetsComponent implements OnInit, OnDestroy {
...
-    interval(this.refreshInterval)
+    this.subs.sink =  interval(this.refreshInterval)
       .pipe(takeWhile(() => this.isAlive)) // only fires when component is alive
       .subscribe(() => {
         this.getAsset();
       });
   }
-
-  public ngOnDestroy(): void {
-    this.isAlive = false;
```
Comment (praveen-garg) [inline src/app/components/core/asset-readings/assets/assets.component.ts]: is this also being taken care of by super?

File: src/app/components/core/backup-restore/backup-restore.component.ts [hunk 6]
```diff
@@ -172,9 +172,4 @@ export class BackupRestoreComponent implements OnInit, OnDestroy {
   public hideLoadingSpinner() {
     this.showSpinner = false;
   }
-
-  ngOnDestroy() {
-    this.isAlive = false;
```
Comment (praveen-garg) [inline src/app/components/core/backup-restore/backup-restore.component.ts:177]: is this being taken care of by super?

File: src/app/components/core/south/south.component.ts [hunk 7]
```diff
@@ -110,10 +111,4 @@ export class SouthComponent implements OnInit, OnDestroy {
   public hideLoadingSpinner() {
     this.showSpinner = false;
   }
-
-  public ngOnDestroy(): void {
-    this.isAlive = false;
```
Comment (praveen-garg) [inline src/app/components/core/south/south.component.ts:115]: is this being taken care of by super?

File: src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts [hunk 8]
```diff
@@ -510,8 +512,4 @@ export class ReadingsGraphComponent implements OnDestroy {
   keyDescOrder = (a: KeyValue<number, string>, b: KeyValue<number, string>): number => {
     return a.key > b.key ? -1 : (b.key > a.key ? 1 : 0);
   }
-
-  public ngOnDestroy(): void {
-    this.isAlive = false;
```
Comment (praveen-garg) [inline src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts:515]: is this being taken care of by super?

**Review by praveen-garg — CHANGES_REQUESTED** [review]
this.isAlive and clearInterval are not being reset on ngOnDestroy. how are these being taken care of now?

File: src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts [hunk 9]
```diff
@@ -110,6 +112,7 @@ export class ReadingsGraphComponent implements OnDestroy {
       activeDropDowns[0].classList.remove('is-active');
     }
     this.optedTime = ASSET_READINGS_TIME_FILTER;
+    super.ngOnDestroy(); // call to unsubscribes all subscriptions
```
Comment (praveen-garg) [inline src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts:115]: I'm slightly worried, this is calling ngOnDestroy of component. While we are just toggling the modal (part of component). May be there are only modal level variables  today but that is component level life-cycle event.

Am I missing anything?

File: src/app/components/core/north/north-task-modal/north-task-modal.component.ts [hunk 10]
```diff
@@ -153,6 +156,7 @@ export class NorthTaskModalComponent implements OnInit, OnChanges {
       this.viewConfigItemComponent.passwordOnChangeFired = false;
       this.viewConfigItemComponent.passwordMatched = true;
     }
+    super.ngOnDestroy(); // call to unsubscribes all subscriptions
```
Comment (praveen-garg) [inline src/app/components/core/north/north-task-modal/north-task-modal.component.ts:159]: I'm slightly worried, this is calling ngOnDestroy of component. While we are just toggling the modal (part of component). May be there are only modal level variables  today but that is component level life-cycle event.

**Review by praveen-garg — CHANGES_REQUESTED** [review]

File: src/app/components/layout/navbar/navbar.component.ts (hunk 1 above)
Comment (shariq1501) [reply src/app/components/layout/navbar/navbar.component.ts:297]: ngOnDestroy() revered now at child component level to fix this issue.

File: src/app/components/core/north/north.component.ts (hunk 4 above)
Comment (shariq1501) [reply src/app/components/core/north/north.component.ts:112]: ngOnDestroy() revered now at child component level to fix this issue.

File: src/app/components/core/north/north-task-modal/north-task-modal.component.ts (hunk 10 above)
Comment (shariq1501) [reply src/app/components/core/north/north-task-modal/north-task-modal.component.ts:159]: Model is a part of a component but as a child component with separate life cycle. The issue is here the modal ngOnDestory() method not getting called when toggling the modal. So for calling that destroy method we need to  explicitly call the ngOnDestroy method

**Comment by shariq1501** [comment]
Close in favour of https://github.com/foglamp/foglamp-gui/pull/548

**Pull request closed without merging**